python/pyspark/sql/connect/session.py

```diff
@@ -418,6 +426,34 @@ def createDataFrame(
             # If no schema supplied by user then get the names of columns only
             if schema is None:
                 _cols = [str(x) if not isinstance(x, str) else x for x in data.columns]
+                session = SparkSession.active()
+                infer_pandas_dict_as_map = (
+                    str(session.conf.get("spark.sql.execution.pandas.inferPandasDictAsMap")).lower()
+                    == "true"
+                )
+                if infer_pandas_dict_as_map:
+                    if data.empty:
+                        raise PySparkValueError(
+                            error_class="CANNOT_INFER_EMPTY_SCHEMA",
+                            message_parameters={},
+                        )
+                    fields = []
+                    pa_schema = pa.Schema.from_pandas(data)
+                    spark_type: Union[MapType, DataType]
+                    for field in pa_schema:
+                        field_type = field.type
+                        if isinstance(field_type, pa.StructType):
+                            pandas_dict_col = data[field.name]
```

Comment:
Instead of looking up the data, can you get the PyArrow type from `pa.StructType.field`? See also https://arrow.apache.org/docs/python/generated/pyarrow.StructType.html

Reply:
Sounds good! Just updated.